lib/lightstep/tracer/transport/http_json.rb

```diff
@@ -0,0 +1,108 @@
+require 'net/http'
+require 'lightstep/tracer/transport/base'
+
+module LightStep
+  module Transport
+    # HTTPJSON is a transport that sends reports via HTTP in JSON format.
+    # It is thread-safe, however it is *not* fork-safe. When forking, all items
+    # in the queue will be copied and sent in duplicate.
+    #
+    # When forking, you should first `disable` the tracer, then `enable` it from
+    # within the fork (and in the parent post-fork). See
+    # `examples/fork_children/main.rb` for an example.
+    class HTTPJSON < Base
+      LIGHTSTEP_HOST = "collector.lightstep.com"
+      LIGHTSTEP_PORT = 443
+      QUEUE_SIZE = 16
+
+      class QueueFullError < LightStep::Error; end
+
+      # Initialize the transport
+      # @param host [String] host of the domain to the endpoind to push data
+      # @param port [Numeric] port on which to connect
+      # @param verbose [Numeric] verbosity level. Right now 0-3 are supported
+      # @param secure [Boolean]
+      # @param access_token [String] access token for LightStep server
+      # @return [HTTPJSON]
+      def initialize(host: LIGHTSTEP_HOST, port: LIGHTSTEP_PORT, verbose: 0, secure: true, access_token:)
+        @host = host
+        @port = port
+        @verbose = verbose
+        @secure = secure
+
+        raise ConfigurationError, "access_token must be a string" unless String === access_token
+        raise ConfigurationError, "access_token cannot be blank"  if access_token.empty?
+        @access_token = access_token
+
+        start_queue
+      end
+
+      # Queue a report for sending
+      def report(report)
+        p report if @verbose >= 3
+        # TODO(ngauthier@gmail.com): the queue could be full here if we're
+        # lagging, which would cause this to block!
+        @queue.push({
+          host: @host,
+          port: @port,
+          secure: @secure,
+          access_token: @access_token,
+          content: report,
+          verbose: @verbose
+        }, true)
+        nil
+      rescue ThreadError
+        raise QueueFullError
+      end
+
+      # Flush the current queue
+      def flush
+        close
```

Comment:
I'm not 100% sure because the testing around high-throughput http isn't really there. I was hoping that your throughput benchmarks could verify it.